### DISCUSSION TITLE
Charge Current Limiter

### DISCUSSION THREAD

**Discussion by Swistunopolis** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#1167] [Ideas]
This might be a bold request, but the relatively new PowerFlow-Feature (absolutely LOVE it!) got me hooked and thinking ;) - what if AlDente could not only limit the maximum SoC of the battery, but also the maximum current on the way?

It's a known fact that lower charging currents keep a battery cooler and are overall better for it's longevity; currently, there is only one solution to achieve this: using a lower-power charging adapter. This, however, has implications: for periods with high system load, it's output may not be enough to sustain the load on its own, rather drawing (significant) energy from the battery.
I'm seeing this with my MacBook Pro M2 Max, which, while idling or doing only light loads, hovers between 8-25 Watts, but, when pushed, can almost max out the OEM 140W Power Supply. However, using this power brick means cooking the battery with 130 Watts worth of charging current every time when plugging it in (until the charge limit is reached).

The "holy grail" in this scenario would be to not only being able to limit the power that can be drawn from the power adapter, but to only limit it for charging purposes (=power going to the battery), while still allowing the system itself to draw more than that.

It's a bold request, I know - but maybe also not?
Would love to hear the opinion of my favorite, fellow Austrian Mac-Software Developers! :)

Kind regards,
Michael

**Comment by MatthiasKerbl** [comment]
Hi @Swistunopolis,

Excellent idea and it makes a lot of sense to add a feature like that. Unfortunately, we have not found a way yet to implement this and don't know if it is possible at all. :/
Also, we already have a feature request open for this here #426.

Best,
Matthias
